### ISSUE TITLE
[RFC] Files mime icons theming 

### ISSUE THREAD

**Issue by skjnldsv** [nextcloud/server#45579]
Since we updated a few mime icons (doc, spreadsheets, pdf, archives...), we stepped away from the themed mime icons to use either org-provided colors (like red for pdf, blue for docx...) or fallback to the standard grey `#969696`

I am not sure we benefit from supporting the svg coloring feature the mime API provides.
As of technical debt, I think we could deprecate and remove it.

![image](https://github.com/nextcloud/server/assets/14975046/61bd6d86-69df-4487-9047-299a5cbd3614)

cc @nextcloud/designers @nextcloud/server-frontend

**Comment by susnux** [comment]
I agree here, but additionally I would suggest to move to an inline-svg approach. Because currently it is not really possible to get the correct primary color.
The color depends on:
- Theme the user uses
- Preferred color schema of the browser

The first is "easy" to get on the backend, the second only with some new browser stuff not supported by e.g. Firefox.

So instead if we move to inline svgs instead of background images, we could simply use `fill: var(--color-primary-element)`.

**Comment by skjnldsv** [comment]
We would never use the primary color. I suggest removing this entirely in the first post :)

**Comment by susnux** [comment]
> We would never use the primary color. I suggest removing this entirely in the first post :)

...

> if folder, we always use the primary color

So we still have the primary for the folder, which suffers this issue on the PHP api

**Comment by skjnldsv** [comment]
No the folder are inline svg right now, we migrated away from using the mime api :)
I'm talking about every other file type but a directory :open_file_folder:

**Comment by jancborchardt** [comment]
The proposal sounds good from the design side. It helps folders, files with previews and specific file types (PDF, docs, spreadsheets, slides) to stand out.
